### ISSUE TITLE
NaN `occurs_loss` (and therefore total loss + gradients) when all samples in a batch are censored

### ISSUE THREAD

**Issue by mmcdermott** [payalchandak/EveryQuery#30]
## Summary

In `EveryQueryModel._forward` (`src/every_query/model.py:655`), `occurs_loss` is computed as a BCE-with-logits loss masked by `~batch.censor`:

```python
censor_logits = self.censor_mlp(query_embed)
censor_loss = self._get_loss(censor_logits, batch.censor, mask=None)

# if future is censored then don't apply loss on whether query occurs
occurs_logits = self.occurs_mlp(query_embed)
occurs_loss = self._get_loss(occurs_logits, batch.occurs, mask=~batch.censor)

loss = censor_loss + occurs_loss
```

When every sample in the batch is censored, `~batch.censor` is all-False, so `_get_loss` applies the mask and hands `BCEWithLogitsLoss` two empty tensors of shape `(0, 1)`. With the default `reduction='mean'`, `BCEWithLogitsLoss(empty, empty) == nan` (0/0).

Verified locally:

```python
>>> import torch
>>> crit = torch.nn.BCEWithLogitsLoss()
>>> loss = crit(torch.zeros((0, 1), requires_grad=True), torch.zeros((0, 1)))
>>> loss
tensor(nan, grad_fn=<BinaryCrossEntropyWithLogitsBackward0>)
>>> torch.isnan(loss).item()
True
```

That NaN then propagates into `loss = censor_loss + occurs_loss`, so the total loss returned by the forward pass is NaN, and `loss.backward()` in `training_step` destroys the optimizer state with NaN gradients.

## When does this trigger in practice?

A batch is "all censored" exactly when every prediction_time in the batch has less than `duration` days of future data. Because:

- Censoring rates grow monotonically with the requested duration (labels are generated for durations 30..365 days).
- `min_context_per_subject` only bounds past context, not future horizon.
- With imbalanced shards or long-duration tasks (e.g., 365 days), per-batch censoring rates of 50–90% are realistic.

So with e.g. 80% censoring and `batch_size=8`, roughly 1 in 6 batches will be fully censored — enough to blow up a training run. The bug is silent: it manifests as NaN loss, the optimizer step NaN-poisons the model, and every subsequent forward produces NaN logits. Lightning's default `detect_anomaly=False` does not surface it.

`_log_metrics` (`lightning_module.py:253-264`) already guards the metrics branch with `if preds.numel() > 0 and target.unique().numel() == 2`, but the loss path has no equivalent guard. That's the inconsistency that makes this a bug rather than a known limitation.

## Severity

**High / critical.** Silent NaN training. No runtime error, no warning, just a corrupted model.

## Proposed fix

Return a zero (detached-from-input but device/dtype-correct) loss when the mask is empty:

```python
def _forward(self, batch: EveryQueryBatch):
    ...
    censor_logits = self.censor_mlp(query_embed)
    censor_loss = self._get_loss(censor_logits, batch.censor, mask=None)

    occurs_logits = self.occurs_mlp(query_embed)
    uncensored_mask = ~batch.censor
    if uncensored_mask.any():
        occurs_loss = self._get_loss(occurs_logits, batch.occurs, mask=uncensored_mask)
    else:
        occurs_loss = occurs_logits.sum() * 0.0  # zero, but differentiable / on the right device+dtype
    loss = censor_loss + occurs_loss
    ...
```

Using `occurs_logits.sum() * 0.0` (or equivalently `torch.zeros((), device=..., dtype=...)`) keeps the loss as a proper 0-dim tensor with the right device/dtype and maintains autograd graph connectivity (no dangling parameter-less loss).

Alternatively, guard inside `_get_loss` itself so any future masked loss head benefits:

```python
def _get_loss(self, logits, target, mask=None):
    target = target.float().unsqueeze(1)
    if mask is not None:
        logits = logits[mask]
        target = target[mask]
        if logits.numel() == 0:
            return logits.sum() * 0.0
    return self.criterion(logits, target)
```

## Test

A regression test should construct a batch where `batch.censor` is all-True and assert:
1. `_forward(batch)` returns a finite total loss.
2. `outputs.occurs_loss` is finite (ideally exactly zero).
3. Calling `loss.backward()` does not introduce NaNs into any `occurs_mlp` parameter.

**Comment by mmcdermott** [comment]
Resolved by #34 (merged in eabd9f3). `_get_loss` now returns a differentiable zero (`logits.sum() * 0.0`) when the masked batch is empty, so all-censored batches produce finite losses and non-NaN gradients. Covered by `TestAllCensoredBatchProducesFiniteLoss` (backward-through-model regression guard) plus a doctest on `_get_loss`.
